### ISSUE TITLE
The loss is very low,but the mAP is 0 when training on my own datasets.

### ISSUE THREAD

**Issue by xinlnix** [xuannianz/EfficientDet#142]
I am training on my own datasets using the code of April version,the loss is very low than previous version, but the mAP is always 0.
Using the same config,Feb. code version has a high loss, but mAP is not zero.
How can I solve the problem

**Comment by xinlnix** [comment]
69/88 [======================>.......] - ETA: 2:02 - loss: 0.8300 - classification_loss: 0.5200 - regression_loss: 0.3101/gdrive/My Drive/project/EfficientDet-200406-anchorfree-official/generators/common.py:165: UserWarning: Image with id 694 (shape (1540, 2048, 3)) contains the following invalid boxes: [[ 9.660e+02 -1.000e+00  1.077e+03  4.400e+01]].
  annotations['bboxes'][invalid_indices, :]
72/88 [=======================>......] - ETA: 1:42 - loss: 0.8117 - classification_loss: 0.5063 - regression_loss: 0.3054/gdrive/My Drive/project/EfficientDet-200406-anchorfree-official/generators/common.py:165: UserWarning: Image with id 613 (shape (1540, 2048, 3)) contains the following invalid boxes: [[ 1.041e+03 -1.000e+00  1.108e+03  2.000e+01]].
  annotations['bboxes'][invalid_indices, :]
74/88 [========================>.....] - ETA: 1:30 - loss: 0.7994 - classification_loss: 0.4971 - regression_loss: 0.3023/gdrive/My Drive/project/EfficientDet-200406-anchorfree-official/generators/common.py:165: UserWarning: Image with id 1044 (shape (1540, 2048, 3)) contains the following invalid boxes: [[ 1.265e+03 -1.000e+00  1.337e+03  2.100e+01]].
  annotations['bboxes'][invalid_indices, :]
75/88 [========================>.....] - ETA: 1:23 - loss: 0.7933 - classification_loss: 0.4926 - regression_loss: 0.3007/gdrive/My Drive/project/EfficientDet-200406-anchorfree-official/generators/common.py:165: UserWarning: Image with id 1133 (shape (1540, 2048, 3)) contains the following invalid boxes: [[ 1.360e+03 -1.000e+00  1.540e+03  4.400e+01]
 [ 1.360e+03 -1.000e+00  1.531e+03  4.100e+01]].
  annotations['bboxes'][invalid_indices, :]
87/88 [============================>.] - ETA: 6s - loss: 0.7331 - classification_loss: 0.4490 - regression_loss: 0.2842 Epoch 1/50
/gdrive/My Drive/project/EfficientDet-200406-anchorfree-official/generators/common.py:165: UserWarning: Image with id 189 (shape (1540, 2048, 3)) contains the following invalid boxes: [[ 1.551e+03 -1.000e+00  1.632e+03  3.800e+01]].
  annotations['bboxes'][invalid_indices, :]
 8/88 [=>............................] - ETA: 5:48 - loss: 1.5559 - classification_loss: 1.1548 - regression_loss: 0.4011/gdrive/My Drive/project/EfficientDet-200406-anchorfree-official/generators/common.py:165: UserWarning: Image with id 395 (shape (1540, 2048, 3)) contains the following invalid boxes: [[ 1.859e+03 -1.000e+00  1.971e+03  2.900e+01]].
  annotations['bboxes'][invalid_indices, :]
10/88 [==>...........................] - ETA: 5:39 - loss: 1.5578 - classification_loss: 1.1551 - regression_loss: 0.4027/gdrive/My Drive/project/EfficientDet-200406-anchorfree-official/generators/common.py:165: UserWarning: Image with id 240 (shape (1540, 2048, 3)) contains the following invalid boxes: [[ 1.913e+03 -1.000e+00  2.003e+03  2.300e+01]].
  annotations['bboxes'][invalid_indices, :]
Running network: N/A% (0 of 522) |        | Elapsed Time: 0:00:00 ETA:  --:--:--/gdrive/My Drive/project/EfficientDet-200406-anchorfree-official/generators/common.py:165: UserWarning: Image with id 110 (shape (1540, 2048, 3)) contains the following invalid boxes: [[609.  -1. 698.  42.]].
  annotations['bboxes'][invalid_indices, :]
Running network: 100% (522 of 522) |######| Elapsed Time: 0:01:04 Time:  0:01:04
Parsing annotations: 100% (522 of 522) |##| Elapsed Time: 0:00:00 Time:  0:00:00
num_fp=52200.0, num_tp=0.0
948 instances of class plate with average precision: 0.0000
WARNING:tensorflow:From /gdrive/My Drive/project/EfficientDet-200406-anchorfree-official/eval/pascal.py:94: The name tf.Summary is deprecated. Please use tf.compat.v1.Summary instead.

mAP: 0.0000

**Comment by chunleiml** [comment]
I have the same problem as you. Have you solved the problem

> 69/88 [======================>.......] - ETA: 2:02 - loss: 0.8300 - classification_loss: 0.5200 - regression_loss: 0.3101/gdrive/My Drive/project/EfficientDet-200406-anchorfree-official/generators/common.py:165: UserWarning: Image with id 694 (shape (1540, 2048, 3)) contains the following invalid boxes: [[ 9.660e+02 -1.000e+00 1.077e+03 4.400e+01]].
> annotations['bboxes'][invalid_indices, :]
> 72/88 [=======================>......] - ETA: 1:42 - loss: 0.8117 - classification_loss: 0.5063 - regression_loss: 0.3054/gdrive/My Drive/project/EfficientDet-200406-anchorfree-official/generators/common.py:165: UserWarning: Image with id 613 (shape (1540, 2048, 3)) contains the following invalid boxes: [[ 1.041e+03 -1.000e+00 1.108e+03 2.000e+01]].
> annotations['bboxes'][invalid_indices, :]
> 74/88 [========================>.....] - ETA: 1:30 - loss: 0.7994 - classification_loss: 0.4971 - regression_loss: 0.3023/gdrive/My Drive/project/EfficientDet-200406-anchorfree-official/generators/common.py:165: UserWarning: Image with id 1044 (shape (1540, 2048, 3)) contains the following invalid boxes: [[ 1.265e+03 -1.000e+00 1.337e+03 2.100e+01]].
> annotations['bboxes'][invalid_indices, :]
> 75/88 [========================>.....] - ETA: 1:23 - loss: 0.7933 - classification_loss: 0.4926 - regression_loss: 0.3007/gdrive/My Drive/project/EfficientDet-200406-anchorfree-official/generators/common.py:165: UserWarning: Image with id 1133 (shape (1540, 2048, 3)) contains the following invalid boxes: [[ 1.360e+03 -1.000e+00 1.540e+03 4.400e+01]
> [ 1.360e+03 -1.000e+00 1.531e+03 4.100e+01]].
> annotations['bboxes'][invalid_indices, :]
> 87/88 [============================>.] - ETA: 6s - loss: 0.7331 - classification_loss: 0.4490 - regression_loss: 0.2842 Epoch 1/50
> /gdrive/My Drive/project/EfficientDet-200406-anchorfree-official/generators/common.py:165: UserWarning: Image with id 189 (shape (1540, 2048, 3)) contains the following invalid boxes: [[ 1.551e+03 -1.000e+00 1.632e+03 3.800e+01]].
> annotations['bboxes'][invalid_indices, :]
> 8/88 [=>............................] - ETA: 5:48 - loss: 1.5559 - classification_loss: 1.1548 - regression_loss: 0.4011/gdrive/My Drive/project/EfficientDet-200406-anchorfree-official/generators/common.py:165: UserWarning: Image with id 395 (shape (1540, 2048, 3)) contains the following invalid boxes: [[ 1.859e+03 -1.000e+00 1.971e+03 2.900e+01]].
> annotations['bboxes'][invalid_indices, :]
> 10/88 [==>...........................] - ETA: 5:39 - loss: 1.5578 - classification_loss: 1.1551 - regression_loss: 0.4027/gdrive/My Drive/project/EfficientDet-200406-anchorfree-official/generators/common.py:165: UserWarning: Image with id 240 (shape (1540, 2048, 3)) contains the following invalid boxes: [[ 1.913e+03 -1.000e+00 2.003e+03 2.300e+01]].
> annotations['bboxes'][invalid_indices, :]
> Running network: N/A% (0 of 522) | | Elapsed Time: 0:00:00 ETA: --:--:--/gdrive/My Drive/project/EfficientDet-200406-anchorfree-official/generators/common.py:165: UserWarning: Image with id 110 (shape (1540, 2048, 3)) contains the following invalid boxes: [[609. -1. 698. 42.]].
> annotations['bboxes'][invalid_indices, :]
> Running network: 100% (522 of 522) |######| Elapsed Time: 0:01:04 Time: 0:01:04
> Parsing annotations: 100% (522 of 522) |##| Elapsed Time: 0:00:00 Time: 0:00:00
> num_fp=52200.0, num_tp=0.0
> 948 instances of class plate with average precision: 0.0000
> WARNING:tensorflow:From /gdrive/My Drive/project/EfficientDet-200406-anchorfree-official/eval/pascal.py:94: The name tf.Summary is deprecated. Please use tf.compat.v1.Summary instead.
> 
> mAP: 0.0000

I have the same problem as you. Have you solved the problem

**Comment by xinlnix** [comment]
> I have the same problem as you. Have you solved the problem
> 
> > 69/88 [======================>.......] - ETA: 2:02 - loss: 0.8300 - classification_loss: 0.5200 - regression_loss: 0.3101/gdrive/My Drive/project/EfficientDet-200406-anchorfree-official/generators/common.py:165: UserWarning: Image with id 694 (shape (1540, 2048, 3)) contains the following invalid boxes: [[ 9.660e+02 -1.000e+00 1.077e+03 4.400e+01]].
> > annotations['bboxes'][invalid_indices, :]
> > 72/88 [=======================>......] - ETA: 1:42 - loss: 0.8117 - classification_loss: 0.5063 - regression_loss: 0.3054/gdrive/My Drive/project/EfficientDet-200406-anchorfree-official/generators/common.py:165: UserWarning: Image with id 613 (shape (1540, 2048, 3)) contains the following invalid boxes: [[ 1.041e+03 -1.000e+00 1.108e+03 2.000e+01]].
> > annotations['bboxes'][invalid_indices, :]
> > 74/88 [========================>.....] - ETA: 1:30 - loss: 0.7994 - classification_loss: 0.4971 - regression_loss: 0.3023/gdrive/My Drive/project/EfficientDet-200406-anchorfree-official/generators/common.py:165: UserWarning: Image with id 1044 (shape (1540, 2048, 3)) contains the following invalid boxes: [[ 1.265e+03 -1.000e+00 1.337e+03 2.100e+01]].
> > annotations['bboxes'][invalid_indices, :]
> > 75/88 [========================>.....] - ETA: 1:23 - loss: 0.7933 - classification_loss: 0.4926 - regression_loss: 0.3007/gdrive/My Drive/project/EfficientDet-200406-anchorfree-official/generators/common.py:165: UserWarning: Image with id 1133 (shape (1540, 2048, 3)) contains the following invalid boxes: [[ 1.360e+03 -1.000e+00 1.540e+03 4.400e+01]
> > [ 1.360e+03 -1.000e+00 1.531e+03 4.100e+01]].
> > annotations['bboxes'][invalid_indices, :]
> > 87/88 [============================>.] - ETA: 6s - loss: 0.7331 - classification_loss: 0.4490 - regression_loss: 0.2842 Epoch 1/50
> > /gdrive/My Drive/project/EfficientDet-200406-anchorfree-official/generators/common.py:165: UserWarning: Image with id 189 (shape (1540, 2048, 3)) contains the following invalid boxes: [[ 1.551e+03 -1.000e+00 1.632e+03 3.800e+01]].
> > annotations['bboxes'][invalid_indices, :]
> > 8/88 [=>............................] - ETA: 5:48 - loss: 1.5559 - classification_loss: 1.1548 - regression_loss: 0.4011/gdrive/My Drive/project/EfficientDet-200406-anchorfree-official/generators/common.py:165: UserWarning: Image with id 395 (shape (1540, 2048, 3)) contains the following invalid boxes: [[ 1.859e+03 -1.000e+00 1.971e+03 2.900e+01]].
> > annotations['bboxes'][invalid_indices, :]
> > 10/88 [==>...........................] - ETA: 5:39 - loss: 1.5578 - classification_loss: 1.1551 - regression_loss: 0.4027/gdrive/My Drive/project/EfficientDet-200406-anchorfree-official/generators/common.py:165: UserWarning: Image with id 240 (shape (1540, 2048, 3)) contains the following invalid boxes: [[ 1.913e+03 -1.000e+00 2.003e+03 2.300e+01]].
> > annotations['bboxes'][invalid_indices, :]
> > Running network: N/A% (0 of 522) | | Elapsed Time: 0:00:00 ETA: --:--:--/gdrive/My Drive/project/EfficientDet-200406-anchorfree-official/generators/common.py:165: UserWarning: Image with id 110 (shape (1540, 2048, 3)) contains the following invalid boxes: [[609. -1. 698. 42.]].
> > annotations['bboxes'][invalid_indices, :]
> > Running network: 100% (522 of 522) |######| Elapsed Time: 0:01:04 Time: 0:01:04
> > Parsing annotations: 100% (522 of 522) |##| Elapsed Time: 0:00:00 Time: 0:00:00
> > num_fp=52200.0, num_tp=0.0
> > 948 instances of class plate with average precision: 0.0000
> > WARNING:tensorflow:From /gdrive/My Drive/project/EfficientDet-200406-anchorfree-official/eval/pascal.py:94: The name tf.Summary is deprecated. Please use tf.compat.v1.Summary instead.
> > mAP: 0.0000
> 
> I have the same problem as you. Have you solved the problem

I am sorry to hear that.Obviously I don not solve the problem.
The same config doesn't work.Can you have any idea about this?

**Comment by CraigWang1** [comment]
Hi, are you running the exact same commands as step 1? If you're in step 2, I found that you have to edit the learning rate from '1e-3' to '1e-4' in train.py if that helps.

**Comment by xinlnix** [comment]
> Hi, are you running the exact same commands as step 1? If you're in step 2, I found that you have to edit the learning rate from '1e-3' to '1e-4' in train.py if that helps.

This is the step 1. And the commands is `!python3 train.py --snapshot imagenet --phi 0 --weighted-bifpn --gpu 0 --random-transform --compute-val-loss --freeze-backbone --batch-size 32 --steps 88 pascal  /content/datasets/VOC2007 `

**Comment by CraigWang1** [comment]
Maybe try taking out `--weighted-bifpn`?

**Comment by xinlnix** [comment]
> Maybe try taking out `--weighted-bifpn`?

Ok, let me try

**Comment by xinlnix** [comment]
> Maybe try taking out `--weighted-bifpn`?

`num_fp=52050.0, num_tp=150.0
948 instances of class plate with average precision: 0.0010
mAP: 0.0010`
It changes.But the mAP is much lower than previous.

**Comment by xinlnix** [comment]
> > Maybe try taking out `--weighted-bifpn`?
> 
> Ok, let me try

the epoch 2 is below
`Running network: 100% (522 of 522) |######| Elapsed Time: 0:00:41 Time:  0:00:41
Parsing annotations: 100% (522 of 522) |##| Elapsed Time: 0:00:00 Time:  0:00:00
num_fp=0, num_tp=0
948 instances of class plate with average precision: 0.0000
mAP: 0.0000
`

**Comment by northeastsquare** [comment]
I also met this, check your dataset's warning:
UserWarning: Image with id 694 (shape (1540, 2048, 3)) contains the following invalid boxes: [[ 9.660e+02 -1.000e+00 1.077e+03 4.400e+01]]

**Comment by xinlnix** [comment]
@northeastsquare Yes,I do have the datasets issue, but I ignore it because I think it do nothing.Can your project work well after fix your datasets?

**Comment by Wojak27** [comment]
Any progress on this issue? I'm having the same problems...
